control-plane/agents/core/src/core/reconciler/pool/mod.rs

```diff
@@ -110,3 +113,45 @@ async fn missing_pool_state_reconciler(
         PollResult::Ok(PollerState::Idle)
     }
 }
+
+/// If a pool is tried to be deleted after its corresponding mayastor node is down,
+/// the pool deletion gets struck in Deleting state, this creates a problem as when
+/// the node comes up we cannot create a pool with same specs, the deleting_pool_spec_reconciler
+/// cleans up any such pool when node comes up.
+#[tracing::instrument(skip(pool_spec, context), fields(pool.uuid = %pool_spec.lock().id))]
+async fn deleting_pool_spec_reconciler(
+    pool_spec: Arc<Mutex<PoolSpec>>,
+    context: &PollContext,
+) -> PollResult {
+    if !pool_spec.lock().status().deleting() {
+        // nothing to do here
+        return PollResult::Ok(PollerState::Idle);
+    }
+    let pool = pool_spec.lock().clone();
+    match context.registry().get_node_wrapper(&pool.node).await {
+        Ok(node) => {
+            if !node.read().await.is_online() {
+                return PollResult::Ok(PollerState::Idle);
+            }
+        }
+        Err(_) => return PollResult::Ok(PollerState::Idle),
+    };
+    let request = DestroyPool {
+        node: pool.clone().node,
+        id: pool.clone().id,
+    };
```

Comment:
Nit: You can clone the individual fields rather than the entire pool structure.

`let request = DestroyPool {
        node: pool.node.clone(),
        id: pool.id.clone(),
    };`

Reply:
Done.